### ISSUE TITLE
npm install rx-flux => not in npm registry

### ISSUE THREAD

**Issue by pillowsoft** [fdecampredon/rx-flux#2]
I'd like to try it out, but get this:

npm ERR! 404 'rx-flux' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it

I can of course pull it off github, but figure you might want to get it in the registry. Thanks!!!

**Comment by hekike** [comment]
@fdecampredon what do you think, how much time do you need? Can I help with something? I really would like to see it on npm ;)

**Comment by fdecampredon** [comment]
in fact I have a last refactoring scheduled for store :d and I should publish it, it should be ready next week :) sorry for the delay.

**Comment by hekike** [comment]
@fdecampredon sounds great! ;)

**Comment by jfsiii** [comment]
@fdecampredon is 96a60e7 the refactoring you mentioned in https://github.com/fdecampredon/rx-flux/issues/2#issuecomment-68537934?

**Comment by fdecampredon** [comment]
Yup now I just need to figure how to handle error properly/update doc and the 0.2 should be published on npm, I also just created a 0.1 tag with old store version for those who used it (if there is any)

**Comment by hekike** [comment]
@fdecampredon why not follow the proper [semver](http://semver.org/) and release 1.0.0? 
Btw I mentioned your lib here: http://blog.risingstack.com/flux-inspired-libraries-with-react/ ;)

**Comment by fdecampredon** [comment]
Thank you @hekike.

Yes I'm thinking about it however I'm still not 100% happy with the current implementation.
Here are my concern : 

* Error handling  in store : 
  * from the initialValue observable
  * in transform function
* Error handling in action: 
  * Should the observable notify error ?
  * Should it rethrows the error ?
* Where data validation should be done: 
 * Inside *actions* ? 
 * inside the react component ?

Also I'm thinking about a way to define action inspired by https://github.com/staltz/cycle (see the actual todomvc example). But I'm not so happy with the result :/.

Finally while I like my operation system on store, and how it allows to be reverted (for smooth integration with remote async call), I'm still not 100% happy with the result, I'm thinking about a possibility to directly listen 'actions' on a node server (yes it might seems crazy but it has some logic from a reactive point of view) but I think it would need to be done in a separate project.

Until all those concerns are cleared I'll publish only 0.x the 0.2 is scheduled soon, I just want to handle error properly. By the way if any of you have feedbacks/comments It could help me in my reflexion

**Comment by ivanoats** [comment]
I'd like to see an 0.x soon please, because:

1) The examples depend on it in npm
2) I'd like to start using this and potentially contributing. I think it's *great*

**Comment by BerkeleyTrue** [comment]
How about adding onError property to the store definition and having it handle both operation errors.

**Comment by fdecampredon** [comment]
OK since there seems to be some bunch of people really interested in `rx-flux` I'm quite happy, however I feel sorry and I must apologize, because I didn't had much time to work on it until now.
Now I want to resume my work on that one and I won't switch to any other side projects until I have designed the right api for this one and published it on npm, rest assured.

However I think I had some good idea here and I think this project deserves the right api before being published, If any of you want to help me please go see #10

**Comment by zbyte64** [comment]
:+1:

**Comment by aexmachina** [comment]
:+1:

**Comment by SupremeTechnopriest** [comment]
@fdecampredon let me know if I can be of any service to you.  For anyone who needs the latest npm module, I published under my own namespace. `npm install @supremetechnopriest/rx-flux` you can use that in the mean time while @fdecampredon and I get this library production ready.

**Comment by fdecampredon** [comment]
I believe that rx-flux offer no advantage over @gaearon https://github.com/rackt/redux, and I clearly encourage you to use it instead of this project. 
I won't maintain it, anyway it always have been more an experiment than anything (that's mainly why I never published it to npm).

**Comment by fdecampredon** [comment]
SupremeTechnopriest I gave you the right over the project, if you wish to maintain it feel free !
Feel also free to ask for any advice!
